src/main/java/com/programmers/ticketparis/controller/schedule/ScheduleController.java

```diff
@@ -0,0 +1,42 @@
+package com.programmers.ticketparis.controller.schedule;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.programmers.ticketparis.dto.ApiResponse;
+import com.programmers.ticketparis.dto.schedule.request.ScheduleCreateRequest;
+import com.programmers.ticketparis.dto.schedule.response.ScheduleResponse;
+import com.programmers.ticketparis.service.schedule.ScheduleService;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/performances/{performanceId}")
+public class ScheduleController {
+
+    private final ScheduleService scheduleService;
+
+    @PostMapping("/schedules")
+    @ResponseStatus(HttpStatus.CREATED)
+    public ApiResponse<ScheduleResponse> createSchedule(@PathVariable Long performanceId, @Valid @RequestBody ScheduleCreateRequest request, HttpServletRequest httpServletRequest) {
+        ScheduleResponse response = scheduleService.createSchedule(performanceId, request);
+        String requestURI = httpServletRequest.getRequestURI();
+
+        return ApiResponse.of(requestURI, response);
```

Comment:
`p3`
ApiResponse에서 path 정보를 클라이언트에게 보내주기 위한 코드 같은데 아무래도 응답 값이 있는 모든 메서드에 적어줘야하다보니 최대한 코드를 줄이는 것도 좋을 것 같습니다! 아래와 같이 ApiResponse에는 httpServletRequest를 매개변수로 넘겨주고 ApiResponse에서 getRequestURI로 가져와서 처리하는 것은 어떻게 생각하시나요?

```suggestion
        return ApiResponse.of(httpServletRequest, response);
```

Reply:
메소드 시그니쳐만 봐도 어떤행위를 하는지 알수 있어야 한다고 생각하는데요.
httpServletRequest 안에 무엇을 가져다 쓰지? 하면 해당 of method를 다시 들어가봐야 하지 않을까 싶기도 합니다.
다만, 지금의 코드에서는 간단한 코드라 아래처럼 조정하면 어떨지 싶네요.

\+ 좀 더 하자면 파라미터가 2개 이상일 경우엔 of보다 빌더가 좋지 앟을까 하는 생각도 들어요.
public XXX of(String a, String b) 이렇게 되면 a 와 b를 바꿔 사용해도 컴파일에선 에러가 안나거등요. 물론 빌더도 마찬가지지만 파라미터를 지정해서 셋팅 해주니 좀더 명확하지 않을까 싶기도 하고요 ㅎ

```java
return ApiResponse.of(httpServletRequest.getRequestURI(), response);
```

Reply:
이런 번거로운 점 때문에 `ResponseBodyAdvice`를 쓰는것 같습니다.
도입을 검토해보세요~!~!~!